src/basic/locale-util.c

```diff
@@ -109,7 +109,8 @@ static int add_locales_from_archive(Set *locales) {
         if (!S_ISREG(st.st_mode))
                 return -EBADMSG;
 
-        if (st.st_size < (off_t) sizeof(struct locarhead))
+        if (st.st_size < (off_t) sizeof(struct locarhead) ||
+            (uintmax_t) st.st_size > SIZE_MAX)
                 return -EBADMSG;
```

Comment:
hmm, I think we should return a different error in that case, i.e. EFBIG or so.

Reply:
done